Bandwidth.Iris/Client.cs

```diff
@@ -396,51 +397,50 @@ private static string FixPath(string path)
-                if (xml.Length > 0)
+                try
                 {
-                    var doc = XDocument.Parse(xml);
-                    var code = doc.Descendants("ErrorCode").FirstOrDefault();
-                    var description = doc.Descendants("Description").FirstOrDefault();
-                    if (code == null)
+                    var xml = await response.Content.ReadAsStringAsync();
+                    if (xml.Length > 0)
                     {
-                        var error = doc.Descendants("Error").FirstOrDefault();
-                        if (error == null)
+                        var doc = XDocument.Parse(xml);
+                        var code = doc.Descendants("ErrorCode").FirstOrDefault();
+                        var description = doc.Descendants("Description").FirstOrDefault();
+                        if (code == null)
                         {
-                            var exceptions =
-                                (from item in doc.Descendants("Errors")
-                                    select
-                                        (Exception) new BandwidthIrisException(item.Element("Code").Value,
-                                            item.Element("Description").Value, response.StatusCode)).ToArray();
-                            if (exceptions.Length > 0)
+                            var error = doc.Descendants("Error").FirstOrDefault();
+                            if (error == null)
+                            {
+                                var exceptions =
+                                    (from item in doc.Descendants("Errors")
+                                     select
+                                         (Exception)new BandwidthIrisException(item.Element("Code").Value,
+                                             item.Element("Description").Value, response.StatusCode)).ToArray();
+                                if (exceptions.Length > 0)
+                                {
+                                    throw new AggregateException(exceptions);
+                                }
+                                code = doc.Descendants("resultCode").FirstOrDefault();
+                                description = doc.Descendants("resultMessage").FirstOrDefault();
+                            }
+                            else
                             {
-                                throw new AggregateException(exceptions);
+                                code = error.Element("Code");
+                                description = error.Element("Description");
                             }
-                            code = doc.Descendants("resultCode").FirstOrDefault();
-                            description = doc.Descendants("resultMessage").FirstOrDefault();
                         }
-                        else
+                        if (code != null && description != null && !string.IsNullOrEmpty(code.Value) && code.Value != "0")
```

Comment:
is `code != null` necessary in this block since the previous check is `code == null`?

Reply:
Probably not 😂